### PR TITLE
Audio recordings/create directory loadurls

### PR DESCRIPTION
**Pull request by JB184351**
I managed to be able to create a Directory and add URLs to it but honestly, I feel like it shouldn't work after investigating for some time and was also able to load all the URLs from said Directory to the TableView.

Items that I need to fix.
* Audio doesn't playback as intended when selecting from the TableView
*  Make the tableView cells text cells shorter so it is nicer.

### CODE REVIEW

**Comment by JB184351** [comment]
I was able to get the playback to appear but the ui wasn't updating because after changing the audioState from the tableView back to the playbackView that audioState changes to stopped and doesn't call the delegate method unless I call it in viewWillAppear for some reason and even with that there are still some other ui fixes I need to do but in general this works.

**Comment by virgil-adumitroae** [comment]
Debug more 🙂
And remove any commented code or logic that you added to short circuit stuff.

Because I already found the issue.

**Comment by JB184351** [comment]
Were you referring to my `didUpdateAudioState` delegate method being commented out?

File: TalkAloud/TalkAloud/ViewControllers/AudioPlayerViewController.swift [hunk 1]
```diff
@@ -24,12 +24,26 @@ class AudioViewController: UIViewController, AudioEngineStateChangeDelegate {
...
+        // Doing this on ViewWillAppear because in the future
+        // when I have the option to delete recording from the
+        // Audio Player View I will know that this check will always
+        // happen
+        if audioManager.isArrayEmpty() {
+            playAudioButton.isEnabled = false
+        } else {
+            playAudioButton.isEnabled = true
+        }
+        didUpdateAudioState(with: AudioEngine.sharedInstance.audioState)
```
Comment (virgil-adumitroae) [inline TalkAloud/TalkAloud/ViewControllers/AudioPlayerViewController.swift]: You are short circuiting stuff, adding code that is not meant to be added. 
This is not the solution.

Why would your class itself call that delegate method? It doesn't make sense.

Your problem is elsewhere.

File: TalkAloud/TalkAloud.xcodeproj/xcuserdata/J-Beng.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist [hunk 2]
```diff
@@ -7,16 +7,237 @@
       <BreakpointProxy
```
Comment (virgil-adumitroae) [inline TalkAloud/TalkAloud.xcodeproj/xcuserdata/J-Beng.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist]: I'm gonna give you a fine if I see this file commited again. 😂

File: TalkAloud/TalkAloud/Model/AudioEngine.swift [hunk 3]
```diff
@@ -60,9 +61,8 @@ class AudioEngine: NSObject {
     }
     
     func play(withFileURL: URL) {
-        if audioPlayer == nil {
-            setupAudioPlayer(fileURL: withFileURL)
-        }
+        // Change logic to play with a URL everytime not only when NIL
```
Comment (virgil-adumitroae) [inline TalkAloud/TalkAloud/Model/AudioEngine.swift]: Add comments when only the code isn't self explanatory.

File: TalkAloud/TalkAloud.xcodeproj/xcuserdata/J-Beng.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist (hunk 2 above)
Comment (JB184351) [reply TalkAloud/TalkAloud.xcodeproj/xcuserdata/J-Beng.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist]: I swear I thought I put a gitignore on these lol

**Review by virgil-adumitroae — APPROVED** [review]

**Pull request merged**